texar/modules/connectors/connectors.py

```diff
@@ -0,0 +1,786 @@
+            }
+
+        Here:
+
+        "activation_fn" : str
+            The activation function applied to the outputs of the MLP
+            transformation layer. Can be a function,
+            or its name or module path.
+        "name" : str
+            Name of the connector.
+        """
+        return {
+            "activation_fn": "texar.core.layers.identity",
+            "name": "reparameterized_stochastic_connector"
+        }
+
+    def forward(self, # type: ignore
+                transform: bool = True,
+                num_samples: Optional[Union[int, torch.Tensor]] = None
+                ) -> Tuple[Any, Any]:
+        r"""Samples from a distribution and optionally performs transformation
+        with an MLP layer.
+        The distribution must be reparameterizable, i.e.,
+        :python:`distribution.reparameterization_type = FULLY_REPARAMETERIZED`.
+
+        Args:
+            transform (bool): Whether to perform MLP transformation of the
+                distribution samples. If ``False``, the structure/shape of a
+                sample must match :attr:`output_size`.
+            num_samples (optional): An ``int`` or ``int`` ``Tensor``.
+                Number of samples to generate. If not given,
+                generate a single sample. Note that if batch size has
+                already been included in :attr:`distribution`'s dimensionality,
+                :attr:`num_samples` should be left as ``None``.
+
+        Returns:
+            A ``tuple`` (output, sample), where
+            - output: A ``Tensor`` or a (nested) ``tuple`` of Tensors with
+              the same structure and size of :attr:`output_size`.
+              The batch dimension equals :attr:`num_samples` if specified,
+              or is determined by the distribution dimensionality.
+            - sample: The sample from the distribution, prior to transformation.
+        Raises:
+            ValueError: If distribution cannot be reparametrized.
+            ValueError: The output does not match :attr:`output_size`.
+        """
+
+        if num_samples:
+            sample = self._dstr.sample([num_samples]) # type: ignore
+        else:
+            sample = self._dstr.sample() # type: ignore
+
+        if self._dstr.event_shape == []: # type: ignore
+            sample = torch.reshape(
+                sample,
+                sample.size() + torch.Size([1]))
+
+        sample = sample.float()
+        if transform:
+            fn_modules = ['torch', 'torch.nn', 'texar.custom']
+            activation_fn = utils.get_function(
+                self.hparams.activation_fn, fn_modules)
+
+            output, linear_layer = _mlp_transform(
+                sample, self._output_size, activation_fn)
+            self._linear_layers.append(linear_layer)
+
+            _assert_same_size(output, self._output_size)
+
+        return (output, sample)
+
+
+class StochasticConnector(ConnectorBase):
```

Comment:
This class has the same problems as above.

Reply:
Checked.